contrib/natss/pkg/dispatcher/dispatcher/dispatcher_test.go

```diff
@@ -74,12 +75,34 @@ func TestMain(m *testing.M) {
 		logger.Fatalf("Cannot start NATSS: %v", err)
 	}
 	defer stopNatss(stanServer)
-
 	// Create and start Dispatcher.
 	s, err = NewDispatcher(natssTestURL, logger.Desugar())
 	if err != nil {
 		logger.Fatalf("Unable to create NATSS dispatcher: %v", err)
 	}
+	// Starting Connect to establish connection with NATS
+	go s.Connect()
+	// forcing to establish connection to NATS before any tests
+	s.connect <- struct{}{}
+	ready := false
+	ticker := time.NewTicker(time.Second * 5)
+	expire := time.NewTimer(time.Second * 120)
+	for !ready {
+		select {
+		case <-ticker.C:
+			s.subscriptionsMux.Lock()
+			if s.natssConn != nil && (*s.natssConn).NatsConn().IsConnected() {
+				ready = true
+			}
+			s.subscriptionsMux.Unlock()
+			continue
+		case <-expire.C:
+			logger.Fatalf("Failed to connect to NATSS!")
+		}
+	}
+	logger.Infof("><SB> (*s.natssConn).NatsConn().IsConnected(): %t", (*s.natssConn).NatsConn().IsConnected())
```

Comment:
These log lines don't look like they should make it in as-is (I'm guessing <SB> is so you can find them quickly with grep).